parteh/PRTGenericMod.F90

```diff
@@ -0,0 +1,1279 @@
+
+  ! We use this parameter as the value for which we check un-initialized values
+  real(r8), parameter :: check_initialized = -8.8e32_r8
+
+
+  ! -------------------------------------------------------------------------------------
+  ! IMPORTANT!
+  ! All species in all organs should be expressed in terms of KILOGRAMS
+  ! All rates of change are expressed in terms of kilorams / day
+  ! This assumption cannot be broken!
+  ! -------------------------------------------------------------------------------------
+  
+  character(len=len_baseunit), parameter :: mass_unit = 'kg'
+  character(len=len_baseunit), parameter :: mass_rate_unit = 'kg/day'
+
+  ! -------------------------------------------------------------------------------------
+  ! Allocation Hypothesis Types
+  ! These should each have their own module
+  ! -------------------------------------------------------------------------------------
+
+  integer, parameter :: prt_carbon_allom_hyp   = 1
+  integer, parameter :: prt_cnp_flex_allom_hyp = 2   ! Still under development
+
+
+  ! -------------------------------------------------------------------------------------
+  ! Organ types
+  ! These are public indices used to map the organs
+  ! in each hypothesis to organs that acknowledged in the calling model
+  ! -------------------------------------------------------------------------------------
+
+  integer, parameter :: num_organ_types = 6
+  integer, parameter :: all_organs    = 0    ! index for all organs
+  integer, parameter :: leaf_organ    = 1    ! index for leaf organs
+  integer, parameter :: fnrt_organ    = 2    ! index for fine-root organs
+  integer, parameter :: sapw_organ    = 3    ! index for sapwood organs
+  integer, parameter :: store_organ   = 4    ! index for storage organs
+  integer, parameter :: repro_organ   = 5    ! index for reproductive organs
+  integer, parameter :: struct_organ  = 6    ! index for structure (dead) organs
+
+  ! -------------------------------------------------------------------------------------
+  ! Species types
+  ! These are public indices used to map the species in each hypothesis
+  ! to the species that are acknowledged in the calling model
+  ! -------------------------------------------------------------------------------------
+
+  integer, parameter :: num_species_types     = 17    ! Total number of unique species
+                                                    ! curently recognized by PARTEH
+                                                    ! should be max index in list below
+
+  ! The following list of unique public indices should be monotonic, and self-explanatory
+  
+  integer, parameter :: all_carbon_species  = 0
+  integer, parameter :: carbon12_species    = 1
+  integer, parameter :: carbon13_species    = 2
+  integer, parameter :: carbon14_species    = 3
+  integer, parameter :: nitrogen_species    = 4
+  integer, parameter :: phosphorous_species = 5
+  integer, parameter :: potassium_species   = 6
```

Comment:
perhaps we should note that indices above 6 here are purely notional for the foreseeable future in case someone greps through the code and actually thinks we're tracking magnesium?

Reply:
Sure, that sounds reasonable. Will address